server/src/test/java/org/elasticsearch/discovery/SeedHostsResolverTests.java

```diff
@@ -92,4 +146,239 @@ public void testResolvesAddressesInBackgroundAndIgnoresConcurrentCalls() throws
+                    },
+                    new TransportAddress(loopbackAddress, 9302)
+                );
+            }
+        };
+        closeables.push(transport);
+        final TransportService transportService =
+            new TransportService(Settings.EMPTY, transport, threadPool, TransportService.NOOP_TRANSPORT_INTERCEPTOR, x -> null, null,
+                Collections.emptySet());
+        closeables.push(transportService);
+        final List<TransportAddress> transportAddresses = SeedHostsResolver.resolveHostsLists(
+            executorService,
+            logger,
+            Collections.singletonList(NetworkAddress.format(loopbackAddress)),
+            10,
+            transportService,
+            TimeValue.timeValueSeconds(30));
+        assertThat(transportAddresses, hasSize(7));
+        final Set<Integer> ports = new HashSet<>();
+        for (final TransportAddress address : transportAddresses) {
+            assertTrue(address.address().getAddress().isLoopbackAddress());
+            ports.add(address.getPort());
+        }
+        assertThat(ports, equalTo(IntStream.range(9303, 9310).mapToObj(m -> m).collect(Collectors.toSet())));
+    }
+
+    public void testUnknownHost() {
+        final Logger logger = mock(Logger.class);
+        final NetworkService networkService = new NetworkService(Collections.emptyList());
+        final String hostname = randomAlphaOfLength(8);
+        final UnknownHostException unknownHostException = new UnknownHostException(hostname);
+        final Transport transport = new MockNioTransport(
+            Settings.EMPTY,
+            Version.CURRENT,
+            threadPool,
+            networkService,
+            PageCacheRecycler.NON_RECYCLING_INSTANCE,
+            new NamedWriteableRegistry(Collections.emptyList()),
+            new NoneCircuitBreakerService()) {
+
+            @Override
+            public BoundTransportAddress boundAddress() {
+                return new BoundTransportAddress(
+                    new TransportAddress[]{new TransportAddress(InetAddress.getLoopbackAddress(), 9300)},
+                    new TransportAddress(InetAddress.getLoopbackAddress(), 9300)
+                );
+            }
+
+            @Override
+            public TransportAddress[] addressesFromString(String address, int perAddressLimit) throws UnknownHostException {
+                throw unknownHostException;
+            }
+
+        };
+        closeables.push(transport);
+
+        final TransportService transportService =
+            new TransportService(Settings.EMPTY, transport, threadPool, TransportService.NOOP_TRANSPORT_INTERCEPTOR, x -> null, null,
+                Collections.emptySet());
+        closeables.push(transportService);
+
+        final List<TransportAddress> transportAddresses = SeedHostsResolver.resolveHostsLists(
+            executorService,
+            logger,
+            Arrays.asList(hostname),
```

Comment:
IntelliJ prefers:

```suggestion
            singletonList(hostname),
```

(with appropriate imports)

Reply:
see 0bf8a12